### PR TITLE
Main vc

### PR DESCRIPTION
**Pull request by swrw90**
Completed setup for Challenge fetching from Core Data and ChallengeVC is able to display a Challenge.

### CODE REVIEW

File: Challenge+CoreDataClass.swift [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        }
+    }
+    
+    /// Fetch Challenge using offset
+    class func fetchRandomChallenge(with fetchRequest: NSFetchRequest<Challenge>, in context: NSManagedObjectContext) -> Challenge? {
+        
+        do {
+            let challenges =  try context.fetch(fetchRequest)
+            let challenge = challenges.randomElement()
+                return challenge
```
Comment (yhozumi) [inline Challenge+CoreDataClass.swift]: This isn't lined up correctly. You should re align everything.

Comment (yhozumi) [reply Challenge+CoreDataClass.swift]: also you can just do `return challenges.randomElement()`

File: Challenge+CoreDataClass.swift [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        do {
+            let challenges = try context.fetch(fetchRequest)
+            return challenges.first
+        } catch {
+            print("error") // switch to assert
+            return nil
+        }
+    }
+    
+    /// Fetch Challenge using offset
```
Comment (yhozumi) [inline Challenge+CoreDataClass.swift]: Docs don't match the what it's doing at all.

File: Challenge+CoreDataClass.swift [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                return challenge
+            } catch {
+                print("error")
+                return nil
+            }
+            
+        }
+
+    
+    class func createRandomChallengeFetchRequest(with context: NSManagedObjectContext) -> NSFetchRequest<Challenge>? {
```
Comment (yhozumi) [inline Challenge+CoreDataClass.swift]: Is this static method being called from anywhere?

File: Challenge+CoreDataClass.swift [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        _ = json.compactMap { challenge(from: $0, in: context) }
+        
+        do {
+            try context.save()
+        } catch {
+            fatalError("Failed to save")
+        }
+    }
+    
+    class func challenge(from json: JSON, in context: NSManagedObjectContext) -> Challenge? {
```
Comment (yhozumi) [inline Challenge+CoreDataClass.swift]: Add documentation what this does.

File: Challenge+CoreDataClass.swift [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+import Foundation
+import CoreData
+
+typealias JSON = [String: Any]
+
+@objc(Challenge)
+public class Challenge: NSManagedObject {
+    
+    
+    class func insertToStore(from json: [JSON], in context: NSManagedObjectContext)  {
```
Comment (yhozumi) [inline Challenge+CoreDataClass.swift]: add documentation

File: Challenge+CoreDataClass.swift (hunk 3 above)
Comment (yhozumi) [reply Challenge+CoreDataClass.swift]: also add docs.

File: EarthLove/ChallengeViewController.swift [hunk 6]
```diff
@@ -7,32 +7,72 @@
 //
 
 import UIKit
+import CoreData
 
 class ChallengeViewController: UIViewController {
```
Comment (yhozumi) [inline EarthLove/ChallengeViewController.swift]: Add docs on what this view controller does

File: EarthLove/ChallengeViewController.swift [hunk 7]
```diff
@@ -7,32 +7,72 @@
...
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+        
+        if let context = managedObjectContext {
+            
+            if let fetchRequest = Challenge.createRandomChallengeFetchRequest(with: context) {
+                
+                if let challenge = Challenge.fetchRandomChallenge(with: fetchRequest, in: context) {
+//                    print(challenge)
```
Comment (yhozumi) [inline EarthLove/ChallengeViewController.swift]: delete this print

File: Challenge+CoreDataClass.swift (hunk 3 above)
Comment (swrw90) [reply Challenge+CoreDataClass.swift]: this will be called from displayNewChallenge func in ChallengeVC, it's written on the challengeButtons branch.

**Pull request merged**